src/client.rs

```diff
@@ -43,6 +45,50 @@ use crate::{
     AstarteAggregate, Error, Interface,
 };
 
+/// Possible errors returned while handling connection messages
+#[non_exhaustive]
+#[derive(thiserror::Error, Debug)]
+pub enum RecvError {
+    /// Connection error, either gRPC or MQTT
+    #[error("connection error, {0:?}")]
+    Connection(#[source] crate::transport::ConnectionRecvError),
+
+    /// Couldn't parse the mapping path.
+    #[error("invalid mapping path '{}'", .0.path())]
+    InvalidEndpoint(#[from] MappingError),
+
+    /// Couldn't find an interface with the given name.
+    #[error("couldn't find interface '{name}'")]
+    InterfaceNotFound {
+        /// Name of the missing interface.
+        name: String,
+    },
+
+    /// Couldn't find missing mapping in the interface.
+    #[error("couldn't find mapping {mapping} in interface {interface}")]
+    MappingNotFound {
+        /// Name of the interface.
+        interface: String,
+        /// Path of the missing mapping.
+        mapping: String,
+    },
```

Comment:
Is `MappingNotFound` an error that can come also from `MapppingError`?

Reply:
No, it is the equivalent of the `MappingNotFounf` variant you can find in the `src/error.rs` file